llvm/lib/CodeGen/MachineFunction.cpp

```diff
@@ -1356,6 +1359,25 @@ unsigned MachineJumpTableInfo::getEntryAlignment(const DataLayout &TD) const {
   llvm_unreachable("Unknown jump table encoding!");
 }
 
+/// getEntryIsSigned - Return true if the load for the jump table index
+/// should use signed extension, false if zero extension (unsigned)
+bool MachineJumpTableInfo::getEntryIsSigned() const {
+  switch (getEntryKind()) {
+  case MachineJumpTableInfo::EK_BlockAddress:
```

Comment:
Is BlockAddress really signed? Same goes for the GP Rel entries. I know that's the existing behavior, but it seems odd to me.